precompiles/avs/IAVSManager.sol

```diff
@@ -143,15 +157,32 @@ interface IAVSManager {
         bytes calldata pubkeyRegistrationMessageHash
     ) external returns (bool success);
 
+    /// @dev operatorSubmitTask ,  this function enables a operator submit a task result.
+    /// @param sender The external address for calling this method.
+    /// @param taskID The id of task.
+    /// @param taskResponse is the task response data..
+    /// @param blsSignature is the operator bls sig info..
+    /// @param taskContractAddress is contract address of task.
+    /// @param phase The phase of the Two-Phase Commit protocol:
+    ///             0 = Prepare phase (commit preparation)
+    ///             1 = Commit phase (final commitment)
+    function operatorSubmitTask(
+        address sender,
+        uint64 taskID,
+        bytes calldata taskResponse,
+        bytes calldata blsSignature,
+        address taskContractAddress,
+        uint8 phase
+    ) external returns (bool success);
 
-
+    /// QUERIES
     /// @dev Returns the pubkey and pubkey hash of an operator
     /// @param operator is the operator for whom the key is being registered
-    function getRegisteredPubkey(string memory operator) external pure returns (bytes memory pubkey);
+    function getRegisteredPubkey(string memory operator) external view returns (bytes memory pubkey);
 
     /// @dev Returns the operators of all opt-in in the current avs
     /// @param avsAddress avs address
-    function getOptInOperators(address avsAddress) external returns (string[] memory operators);
+    function getOptInOperators(address avsAddress) external view returns (string[] memory operators);
```

Comment:
Should this be `string[]` or `address[]` ? The former looks better to users while the latter is better suited for Solidity operations.

Reply:
The returned address data starts with exo, so it is better to use a string array

Reply:
Each `exo1` address can be converted to `0x` address by using `sdk.AccAddress(hexAddress.Bytes())` and vice-versa by using `common.Address(accAddress.Bytes())`. Solidity has much better support for manipulating the `0x` addresses while it cannot do much with the `exo1` string addresses.

Given that this function is [not just events intended for external consumption](https://github.com/ExocoreNetwork/exocore/pull/220#discussion_r1820021539), I think it would make sense for the function to return the well-supported native Solidity `address` type. Outside of Solidity, it can be converted to either format easily; however, such conversions are not possible within Solidity.

CC @adu-web3: any thoughts?

Reply:
yeah, `exo1` prefix is added when a normal evm address is encoded to its bech32 representation. So how would we use these operator addresses? most of our precompile functions treat operator address as hex string and assume it would be passed from some EOA caller, so it might be consistent with these functions if we assume the operator addresses returned would be passed into these functions as arguments

Reply:
Closing for now. Agree that the bech32-encoded address is more consistent with the rest of the system; however, it would perhaps be a good idea to keep an open mind about changing this later.